org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/WorkspaceManager.xtend

```diff
@@ -163,13 +163,16 @@ import org.eclipse.xtext.workspace.IWorkspaceConfig
     }
     
     protected def boolean exists(URI uri) {
-        getProjectManager(uri).resourceSet.URIConverter.exists(uri, null)
+        getProjectManager(uri)?.resourceSet.URIConverter.exists(uri, null)
     }
     
     def <T> T doRead(URI uri, (Document, XtextResource)=>T work) {
     	val resourceURI = uri.trimFragment
     	val projectMnr = getProjectManager(resourceURI)
-        val resource = projectMnr.getResource(resourceURI) as XtextResource
+        val resource = projectMnr?.getResource(resourceURI) as XtextResource
+        if (resource === null) {
+        	return work.apply(null, null)	
```

Comment:
In our project resources can link against a static library that is not described as a project and it is fine as a user should not be able to access resources from the library. But this code fails with an NPE for a project, e.g. during the computation of locations from language services. Instead of changing all services, I prefer to fix it here. I can override and copy this method to our project. But will not it be useful for other projects?